tools/clang/lib/CodeGen/CGHLSLMS.cpp

```diff
@@ -1656,14 +1656,24 @@ void CGMSHLSLRuntime::AddHLSLFunctionInfo(Function *F, const FunctionDecl *FD) {
       funcProps->ShaderProps.GS.instanceCount = 1;
   }
 
-  // Populate numThreads
-  if (const HLSLNumThreadsAttr *Attr = FD->getAttr<HLSLNumThreadsAttr>()) {
-    if (!(isMS || isAS)) {
-      // Compute and/or node shader
-      if (isCS || funcProps->shaderKind == DXIL::ShaderKind::Invalid)
-        funcProps->shaderKind = DXIL::ShaderKind::Compute;
+  // set isCs
+  if (const HLSLShaderAttr *Attr = FD->getAttr<HLSLShaderAttr>()) {
+    if (Attr->getStage() == "compute") {
+      funcProps->shaderKind = DXIL::ShaderKind::Compute;
       isCS = true;
+    } 
+    else if (Attr->getStage() == "node") {
+      const HLSLShaderAttr *Attr2 = FD->getAttr<HLSLShaderAttr>();
+      if (Attr2 && Attr2->getStage() == "compute")  {
+        funcProps->shaderKind = DXIL::ShaderKind::Compute;
+        isCS = true;
+      }
```

Comment:
Just as an FYI, this isn't going to work. `getAttr` will return only the first atrribute of that type. Calling it twice won't give you the next attr. You'll need to do something like on https://github.com/microsoft/DirectXShaderCompiler/blob/040e4078b3525bccef0b53464b17e0b49d518c1b/tools/clang/lib/CodeGen/CGHLSLMS.cpp#L1574  

In fact, if you follow the lambda functions down, that's the place that isCS and friends gets set in the first place. So all that is needed here for isCS to be correct is to not overwrite it just because we bumped into a numthreads attribute. I think Tex cleaned this up a while ago, but that stray assignment got left behind.